src/opset/Shape.c

```diff
@@ -22,8 +22,15 @@ int Shape(connx_Graph* graph, __attribute__((unused)) uint32_t output_count, uin
           __attribute__((unused)) uint32_t input_count, uint32_t* inputs, __attribute__((unused)) void** attributes) {
     connx_Tensor* data = connx_Graph_get(graph, inputs[0]);
 
-    int32_t end = *(int32_t*)attributes[0];
-    int32_t start = *(int32_t*)attributes[1];
+    int32_t end = data->ndim;
+    int32_t start = 0;
+
+    if (attributes[0] != NULL) {
+        end = *(int32_t*)attributes[0];
+    }
+    if (attributes[1] != NULL) {
```

Comment:
start is NOT optional attribute but default attribute. So you don't need to check attributes[1] is NULL or not. Just use it. We need to trust onnx-connx compiler.